sale_manual_delivery/models/sale_order_line.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2017 Denis Leemann, Camptocamp SA
+# Copyright 2021 Iván Todorovich, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    qty_procured = fields.Float(
+        string="Quantity Procured",
+        help="Quantity already planned or shipped (stock movements already created)",
+        compute="_compute_qty_procured",
+        readonly=True,
+        store=True,
+    )
+    qty_to_procure = fields.Float(
+        string="Quantity to Procure",
+        help="There is Pending qty to add to a delivery",
+        compute="_compute_qty_to_procure",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends(
+        "move_ids.state",
+        "move_ids.scrapped",
+        "move_ids.product_uom_qty",
+        "move_ids.product_uom",
+        "move_ids.location_id",
+        "move_ids.location_dest_id",
+    )
+    def _compute_qty_procured(self):
+        """
+        Computes the already planned quantities for the given sale order lines,
+        based on the existing stock.moves
+        """
+        for line in self:
+            if line.qty_delivered_method == "stock_move":
+                line.qty_procured = line._get_qty_procurement()
```

Comment:
do we need to set 
`line.qty_procured = 0.0`
to cleaunp previously setup data?

Reply:
Done! Thanks!